<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>2 Microbiome Demo Datasets | IMAP: Integrated Microbiome Analysis Pipelines</title>
<meta name="author" content="Teresia Mrema-Buza">
<meta name="description" content="In this section, we provide demo datasets for microbiome analysis. These datasets have been preprocessed and transformed in previous sessions, and now we are preparing to analyze them. We begin by...">
<meta name="generator" content="bookdown 0.37 with bs4_book()">
<meta property="og:title" content="2 Microbiome Demo Datasets | IMAP: Integrated Microbiome Analysis Pipelines">
<meta property="og:type" content="book">
<meta property="og:url" content="https://tmbuza.github.io/imap-data-exploration/demo-datasets.html">
<meta property="og:image" content="https://tmbuza.github.io/imap-data-exploration/images/planning.png">
<meta property="og:description" content="In this section, we provide demo datasets for microbiome analysis. These datasets have been preprocessed and transformed in previous sessions, and now we are preparing to analyze them. We begin by...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="2 Microbiome Demo Datasets | IMAP: Integrated Microbiome Analysis Pipelines">
<meta name="twitter:description" content="In this section, we provide demo datasets for microbiome analysis. These datasets have been preprocessed and transformed in previous sessions, and now we are preparing to analyze them. We begin by...">
<meta name="twitter:image" content="https://tmbuza.github.io/imap-data-exploration/images/planning.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/Sen-0.4.8/font.css" rel="stylesheet">
<link href="https://fonts.googleapis.com/css2?family=DM%20Mono&amp;display=swap" rel="stylesheet">
<link href="libs/Bitter-0.4.8/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.6.1/transition.js"></script><script src="libs/bs3compat-0.6.1/tabs.js"></script><script src="libs/bs3compat-0.6.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="End-to-End Practical User Guides Using Integrated Approaches"><big>IMAP:</big> Integrated Microbiome Analysis Pipelines</a>:
        <small class="text-muted">End-to-End Practical User Guides Using Integrated Approaches</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">IMAP-Part 08: Microbiome Data Exploration Guide</a></li>
<li class="book-part">DATA EXPLORATION</li>
<li><a class="" href="preface.html"><span class="header-section-number">1</span> Exploratory Data Analysis (EDA) of Microbiome Communities</a></li>
<li class="book-part">DEMO DATASET</li>
<li><a class="active" href="demo-datasets.html"><span class="header-section-number">2</span> Microbiome Demo Datasets</a></li>
<li class="book-part">START EXPLORING</li>
<li><a class="" href="exploration-of-visualization-types.html"><span class="header-section-number">3</span> Exploration of Visualization Types</a></li>
<li class="book-part">BAR CHARTS</li>
<li><a class="" href="bar-plots.html"><span class="header-section-number">4</span> Bar Plots</a></li>
<li class="book-part">DIFFERENTIAL ABUNDANCE</li>
<li><a class="" href="differential-analysis-of-microbiome-data.html"><span class="header-section-number">5</span> Differential Analysis of Microbiome Data</a></li>
<li class="book-part">ORDINATION</li>
<li><a class="" href="ordination-plots.html"><span class="header-section-number">6</span> Ordination Plots</a></li>
<li class="book-part">RELATIONSHIP ANALYSIS</li>
<li><a class="" href="patterns-and-associations-analysis.html"><span class="header-section-number">7</span> Patterns and Associations Analysis</a></li>
<li class="book-part">PHYLOGENY ANALYSIS</li>
<li><a class="" href="phylogenetic-analysis.html"><span class="header-section-number">8</span> Phylogenetic Analysis</a></li>
<li class="book-part">APPENDIX</li>
<li><a class="" href="imap-github-repos.html"><span class="header-section-number">A</span> IMAP GitHub Repos</a></li>
<li><a class="" href="session-information.html"><span class="header-section-number">B</span> Session Information</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/tmbuza/imap-data-exploration.git">View book source <i class="fas fa-air-freshener"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="demo-datasets" class="section level1" number="2">
<h1>
<span class="header-section-number">2</span> Microbiome Demo Datasets<a class="anchor" aria-label="anchor" href="#demo-datasets"><i class="fas fa-link"></i></a>
</h1>
<p>In this section, we provide demo datasets for microbiome analysis. These datasets have been preprocessed and transformed in previous sessions, and now we are preparing to analyze them. We begin by importing the R and Phyloseq objects.</p>
<div id="import-preprepared-data" class="section level2" number="2.1">
<h2>
<span class="header-section-number">2.1</span> Import preprepared data<a class="anchor" aria-label="anchor" href="#import-preprepared-data"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="demo-datasets.html#cb1-1" tabindex="-1"></a><span class="co"># Load required packages</span></span>
<span id="cb1-2"><a href="demo-datasets.html#cb1-2" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb1-3"><a href="demo-datasets.html#cb1-3" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-4"><a href="demo-datasets.html#cb1-4" tabindex="-1"></a></span>
<span id="cb1-5"><a href="demo-datasets.html#cb1-5" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">Saved RData objects</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb1-6"><a href="demo-datasets.html#cb1-6" tabindex="-1"></a></span>
<span id="cb1-7"><a href="demo-datasets.html#cb1-7" tabindex="-1"></a>Saved RData objects</span>
<span id="cb1-8"><a href="demo-datasets.html#cb1-8" tabindex="-1"></a></span>
<span id="cb1-9"><a href="demo-datasets.html#cb1-9" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"data/ps_transformed.rda"</span>, <span class="at">verbose =</span> T)</span>
<span id="cb1-10"><a href="demo-datasets.html#cb1-10" tabindex="-1"></a>Loading objects<span class="sc">:</span></span>
<span id="cb1-11"><a href="demo-datasets.html#cb1-11" tabindex="-1"></a>  ps_asin</span>
<span id="cb1-12"><a href="demo-datasets.html#cb1-12" tabindex="-1"></a>  ps_identity</span>
<span id="cb1-13"><a href="demo-datasets.html#cb1-13" tabindex="-1"></a>  ps_compositional</span>
<span id="cb1-14"><a href="demo-datasets.html#cb1-14" tabindex="-1"></a>  ps_z_otu</span>
<span id="cb1-15"><a href="demo-datasets.html#cb1-15" tabindex="-1"></a>  ps_z_sample</span>
<span id="cb1-16"><a href="demo-datasets.html#cb1-16" tabindex="-1"></a>  ps_log10</span>
<span id="cb1-17"><a href="demo-datasets.html#cb1-17" tabindex="-1"></a>  ps_log10p</span>
<span id="cb1-18"><a href="demo-datasets.html#cb1-18" tabindex="-1"></a>  ps_clr</span>
<span id="cb1-19"><a href="demo-datasets.html#cb1-19" tabindex="-1"></a>  ps_shift</span>
<span id="cb1-20"><a href="demo-datasets.html#cb1-20" tabindex="-1"></a>  ps_scale</span>
<span id="cb1-21"><a href="demo-datasets.html#cb1-21" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"data/bray_distances.rda"</span>, <span class="at">verbose =</span> T)</span>
<span id="cb1-22"><a href="demo-datasets.html#cb1-22" tabindex="-1"></a>Loading objects<span class="sc">:</span></span>
<span id="cb1-23"><a href="demo-datasets.html#cb1-23" tabindex="-1"></a>  ps_asin_bray_dist</span>
<span id="cb1-24"><a href="demo-datasets.html#cb1-24" tabindex="-1"></a>  ps_compositional_bray_dist</span>
<span id="cb1-25"><a href="demo-datasets.html#cb1-25" tabindex="-1"></a>  ps_z_otu_bray_dist</span>
<span id="cb1-26"><a href="demo-datasets.html#cb1-26" tabindex="-1"></a>  ps_z_sample_bray_dist</span>
<span id="cb1-27"><a href="demo-datasets.html#cb1-27" tabindex="-1"></a>  ps_log10_bray_dist</span>
<span id="cb1-28"><a href="demo-datasets.html#cb1-28" tabindex="-1"></a>  ps_log10p_bray_dist</span>
<span id="cb1-29"><a href="demo-datasets.html#cb1-29" tabindex="-1"></a>  ps_clr_bray_dist</span>
<span id="cb1-30"><a href="demo-datasets.html#cb1-30" tabindex="-1"></a>  ps_shift_bray_dist</span>
<span id="cb1-31"><a href="demo-datasets.html#cb1-31" tabindex="-1"></a>  ps_scale_bray_dist</span>
<span id="cb1-32"><a href="demo-datasets.html#cb1-32" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"data/reduced_dimension.rda"</span>, <span class="at">verbose =</span> T)</span>
<span id="cb1-33"><a href="demo-datasets.html#cb1-33" tabindex="-1"></a>Loading objects<span class="sc">:</span></span>
<span id="cb1-34"><a href="demo-datasets.html#cb1-34" tabindex="-1"></a>  pca_asin_bray_metrics</span>
<span id="cb1-35"><a href="demo-datasets.html#cb1-35" tabindex="-1"></a>  mds_asin_bray_metrics</span>
<span id="cb1-36"><a href="demo-datasets.html#cb1-36" tabindex="-1"></a>  pcoa_asin_bray_metrics</span>
<span id="cb1-37"><a href="demo-datasets.html#cb1-37" tabindex="-1"></a>  tsne_asin_bray_metrics</span>
<span id="cb1-38"><a href="demo-datasets.html#cb1-38" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"data/phyloseq_raw_rel_psextra_df_objects.rda"</span>, <span class="at">verbose =</span> T)</span>
<span id="cb1-39"><a href="demo-datasets.html#cb1-39" tabindex="-1"></a>Loading objects<span class="sc">:</span></span>
<span id="cb1-40"><a href="demo-datasets.html#cb1-40" tabindex="-1"></a>  ps_raw</span>
<span id="cb1-41"><a href="demo-datasets.html#cb1-41" tabindex="-1"></a>  ps_rel</span>
<span id="cb1-42"><a href="demo-datasets.html#cb1-42" tabindex="-1"></a>  psextra_raw</span>
<span id="cb1-43"><a href="demo-datasets.html#cb1-43" tabindex="-1"></a>  psextra_rel</span>
<span id="cb1-44"><a href="demo-datasets.html#cb1-44" tabindex="-1"></a>  ps_df</span></code></pre></div>
</div>
<div id="demo-data-for-qiime2-analysis" class="section level2" number="2.2">
<h2>
<span class="header-section-number">2.2</span> Demo data for QIIME2 analysis<a class="anchor" aria-label="anchor" href="#demo-data-for-qiime2-analysis"><i class="fas fa-link"></i></a>
</h2>
<p>The demo data for QIIME2 analysis is downloaded and stored in the <code>data/qiime2</code> directory. You can use these files to practice QIIME2 workflows and analysis techniques.</p>
<p>The data was sourced from the QIIME2 tutorials available at <a href="https://docs.qiime2.org/2018.4/data/tutorials/moving-pictures/">https://docs.qiime2.org/2024.4/data/tutorials/moving-pictures/</a>.</p>
<p>It contains the following files:</p>
<ul>
<li>Sample metadata</li>
<li>Feature table</li>
<li>Taxonomy</li>
<li>Shannon vector</li>
<li>Rooted tree</li>
</ul>
<p>This data is essential for leveraging downstream analysis aimed at understanding data analyzed using the QIIME2 pipeline.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files2.html">dir.exists</a></span><span class="op">(</span><span class="st">"data"</span><span class="op">)</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/files2.html">dir.create</a></span><span class="op">(</span><span class="st">"data"</span><span class="op">)</span><span class="op">}</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files2.html">dir.exists</a></span><span class="op">(</span><span class="st">"data/qiime2"</span><span class="op">)</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/files2.html">dir.create</a></span><span class="op">(</span><span class="st">"data/qiime2"</span><span class="op">)</span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span><span class="op">(</span><span class="st">"https://docs.qiime2.org/2024.2/data/tutorials/moving-pictures/table.qza"</span>, <span class="st">"data/qiime2/feature_table.qza"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span><span class="op">(</span><span class="st">"https://data.qiime2.org/2024.2/tutorials/moving-pictures/sample_metadata.tsv"</span>, <span class="st">"data/qiime2/sample_metadata.tsv"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span><span class="op">(</span><span class="st">"https://docs.qiime2.org/2024.2/data/tutorials/moving-pictures/taxonomy.qza"</span>, <span class="st">"data/qiime2/taxonomy.qza"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span><span class="op">(</span><span class="st">"https://docs.qiime2.org/2024.2/data/tutorials/moving-pictures/rooted-tree.qza"</span>, <span class="st">"data/qiime2/rooted_tree.qza"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span><span class="op">(</span><span class="st">"https://docs.qiime2.org/2024.2/data/tutorials/moving-pictures/core-metrics-results/shannon_vector.qza"</span>, <span class="st">"data/qiime2/shannon_vector.qza"</span><span class="op">)</span></span></code></pre></div>

</div>
</div>



  <div class="chapter-nav">
<div class="prev"><a href="preface.html"><span class="header-section-number">1</span> Exploratory Data Analysis (EDA) of Microbiome Communities</a></div>
<div class="next"><a href="exploration-of-visualization-types.html"><span class="header-section-number">3</span> Exploration of Visualization Types</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#demo-datasets"><span class="header-section-number">2</span> Microbiome Demo Datasets</a></li>
<li><a class="nav-link" href="#import-preprepared-data"><span class="header-section-number">2.1</span> Import preprepared data</a></li>
<li><a class="nav-link" href="#demo-data-for-qiime2-analysis"><span class="header-section-number">2.2</span> Demo data for QIIME2 analysis</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/tmbuza/imap-data-exploration.git/blob/gh-pages/01_demo_data_overview.Rmd">View source <i class="fas fa-air-freshener"></i></a></li>
          <li><a id="book-edit" href="https://github.com/tmbuza/imap-data-exploration.git/edit/gh-pages/01_demo_data_overview.Rmd">Edit this page <i class="fas fa-air-freshener"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container">
    <div class="row">
      <div class="col-12 col-md-6 mt-3">
        <p><strong><u><a class="text-light" href="https://microbiome.complexdatainsights.com"><big>IMAP:</big> Integrated Microbiome Analysis Pipelines</a></u></strong>: End-to-End Practical User Guides Using Integrated Approaches</p>
<br><p>Authored by Teresia Mrema-Buza, Last updated: <b>2024-04-26</b>.</p>
<br>
</div>
      <div class="col-12 col-md-6 mt-3">
        <p>Note: This practical guide was created using the <a class="text-light" href="https://bookdown.org/yihui/bookdown/">bookdown</a> R package, styled with the modern bs4_book format.</p>
      </div>

    </div>
  </div>
</footer>
</body>
</html>
