<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>8 Exploration of processded data | IMAP: Integrated Microbiome Analysis Pipelines</title>
<meta name="author" content="Teresia Mrema-Buza">
<meta name="description" content="8.1 Import data library(tidyverse) ## ── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ── ## ✔ dplyr   1.1.4   ✔ readr   2.1.5 ## ✔ forcats  1.0.0   ✔ stringr  1.5.1...">
<meta name="generator" content="bookdown 0.37 with bs4_book()">
<meta property="og:title" content="8 Exploration of processded data | IMAP: Integrated Microbiome Analysis Pipelines">
<meta property="og:type" content="book">
<meta property="og:url" content="https://tmbuza.github.io/imap-data-preparation/exploration-of-processded-data.html">
<meta property="og:image" content="https://tmbuza.github.io/imap-data-preparation/images/planning.png">
<meta property="og:description" content="8.1 Import data library(tidyverse) ## ── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ── ## ✔ dplyr   1.1.4   ✔ readr   2.1.5 ## ✔ forcats  1.0.0   ✔ stringr  1.5.1...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="8 Exploration of processded data | IMAP: Integrated Microbiome Analysis Pipelines">
<meta name="twitter:description" content="8.1 Import data library(tidyverse) ## ── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ── ## ✔ dplyr   1.1.4   ✔ readr   2.1.5 ## ✔ forcats  1.0.0   ✔ stringr  1.5.1...">
<meta name="twitter:image" content="https://tmbuza.github.io/imap-data-preparation/images/planning.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/Sen-0.4.8/font.css" rel="stylesheet">
<link href="https://fonts.googleapis.com/css2?family=DM%20Mono&amp;display=swap" rel="stylesheet">
<link href="libs/Bitter-0.4.8/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.6.1/transition.js"></script><script src="libs/bs3compat-0.6.1/tabs.js"></script><script src="libs/bs3compat-0.6.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="End-to-End Practical User Guides Using Integrated Approaches"><big>IMAP:</big> Integrated Microbiome Analysis Pipelines</a>:
        <small class="text-muted">End-to-End Practical User Guides Using Integrated Approaches</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">IMAP-Part 08: Microbiome Data Exploration Guide</a></li>
<li class="book-part">DATA EXPLORATION</li>
<li><a class="" href="preface.html"><span class="header-section-number">1</span> Exploring Microbial Composition</a></li>
<li class="book-part">EXPLORATION STEPS</li>
<li><a class="" href="getting-started-with-data-exploration.html"><span class="header-section-number">2</span> Getting Started with Data Exploration</a></li>
<li><a class="" href="data-exploration-techniques.html"><span class="header-section-number">3</span> Data Exploration Techniques</a></li>
<li><a class="" href="visualizations-with-r-and-python.html"><span class="header-section-number">4</span> Visualizations with R and Python:</a></li>
<li><a class="" href="advanced-snakemake-features.html"><span class="header-section-number">5</span> Advanced Snakemake Features</a></li>
<li><a class="" href="case-studies-and-examples.html"><span class="header-section-number">6</span> Case Studies and Examples</a></li>
<li><a class="" href="conclusion.html"><span class="header-section-number">7</span> Conclusion</a></li>
<li><a class="active" href="exploration-of-processded-data.html"><span class="header-section-number">8</span> Exploration of processded data</a></li>
<li class="book-part">APPENDIX</li>
<li><a class="" href="imap-github-repos.html"><span class="header-section-number">A</span> IMAP GitHub Repos</a></li>
<li><a class="" href="data-exploration-rulegraph.html"><span class="header-section-number">B</span> Data Exploration Rulegraph</a></li>
<li><a class="" href="session-information.html"><span class="header-section-number">C</span> Session Information</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/tmbuza/imap-data-exploration.git">View book source <i class="fas fa-air-freshener"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="exploration-of-processded-data" class="section level1" number="8">
<h1>
<span class="header-section-number">8</span> Exploration of processded data<a class="anchor" aria-label="anchor" href="#exploration-of-processded-data"><i class="fas fa-link"></i></a>
</h1>
<div id="import-data" class="section level2" number="8.1">
<h2>
<span class="header-section-number">8.1</span> Import data<a class="anchor" aria-label="anchor" href="#import-data"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span></span></code></pre></div>
<pre><code>## ── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
## ✔ dplyr     1.1.4     ✔ readr     2.1.5
## ✔ forcats   1.0.0     ✔ stringr   1.5.1
## ✔ ggplot2   3.4.4     ✔ tibble    3.2.1
## ✔ lubridate 1.9.3     ✔ tidyr     1.3.1
## ✔ purrr     1.0.2     
## ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
## ✖ dplyr::filter() masks stats::filter()
## ✖ dplyr::lag()    masks stats::lag()
## ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217">phyloseq</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://microbiome.github.io/microbiome">microbiome</a></span><span class="op">)</span></span></code></pre></div>
<pre><code>## 
## microbiome R package (microbiome.github.com)
##     
## 
## 
##  Copyright (C) 2011-2022 Leo Lahti, 
##     Sudarshan Shetty et al. &lt;microbiome.github.io&gt;
## 
## 
## Attaching package: 'microbiome'
## 
## The following object is masked from 'package:ggplot2':
## 
##     alpha
## 
## The following object is masked from 'package:base':
## 
##     transform</code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/yiluheihei/microbiomeMarker">microbiomeMarker</a></span><span class="op">)</span></span></code></pre></div>
<pre><code>## Registered S3 method overwritten by 'gplots':
##   method         from     
##   reorder.factor DescTools
## 
## Attaching package: 'microbiomeMarker'
## 
## The following objects are masked from 'package:microbiome':
## 
##     abundances, aggregate_taxa
## 
## The following object is masked from 'package:phyloseq':
## 
##     plot_heatmap</code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"../imap-data-preparation/data/external/external_ps_objects.rda"</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code>## Loading objects:
##   df_GlobalPatterns
##   df_ibd_phylo
##   df_dietswap
##   df_caporaso
##   df_kostic_crc
##   ps_GlobalPatterns
##   ps_ibd_phylo
##   ps_dietswap
##   ps_caporaso
##   ps_kostic_crc</code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"../imap-data-preparation/data/mothur/mothur_phyloseq_objects.rda"</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code>## Loading objects:
##   mtps_metadata
##   mtps_otutable
##   mtps_taxonomy
##   mtps_phylotree
##   ps_tree
##   ps_raw
##   ps_rel</code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"../imap-data-preparation/data/qiime2/qiime2_phyloseq_objects.rda"</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code>## Loading objects:
##   ps_tree
##   ps_raw
##   ps_rel
##   ps_df_raw
##   ps_df_rel</code></pre>
</div>
<div id="dataset-ps_globalpatterns" class="section level2" number="8.2">
<h2>
<span class="header-section-number">8.2</span> Dataset: <code>ps_GlobalPatterns</code><a class="anchor" aria-label="anchor" href="#dataset-ps_globalpatterns"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ps_GlobalPatterns</span></span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 19216 taxa and 26 samples ]
## sample_data() Sample Data:       [ 26 samples by 7 sample variables ]
## tax_table()   Taxonomy Table:    [ 19216 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 19216 tips and 19215 internal nodes ]</code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\nSample Variables\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code>## 
## Sample Variables</code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_variables.html">sample_variables</a></span><span class="op">(</span><span class="va">ps_GlobalPatterns</span><span class="op">)</span></span></code></pre></div>
<pre><code>## [1] "X.SampleID"               "Primer"                  
## [3] "Final_Barcode"            "Barcode_truncated_plus_T"
## [5] "Barcode_full_length"      "SampleType"              
## [7] "Description"</code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\nNumber of levels in selected group\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code>## 
## Number of levels in selected group</code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps_GlobalPatterns</span><span class="op">)</span><span class="op">$</span><span class="va">SampleType</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>## [1] 1 2 3 4 5 6 7 8 9</code></pre>
</div>
<div id="dataset-ps_ibd_phylo" class="section level2" number="8.3">
<h2>
<span class="header-section-number">8.3</span> Dataset: <code>ps_ibd_phylo</code><a class="anchor" aria-label="anchor" href="#dataset-ps_ibd_phylo"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ps_ibd_phylo</span></span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 36349 taxa and 91 samples ]
## sample_data() Sample Data:       [ 91 samples by 15 sample variables ]
## tax_table()   Taxonomy Table:    [ 36349 taxa by 7 taxonomic ranks ]</code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\nSample Variables\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code>## 
## Sample Variables</code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_variables.html">sample_variables</a></span><span class="op">(</span><span class="va">ps_ibd_phylo</span><span class="op">)</span></span></code></pre></div>
<pre><code>##  [1] "sample"       "gender"       "age"          "DiseaseState" "steroids"    
##  [6] "imsp"         "abx"          "mesalamine"   "ibd"          "activity"    
## [11] "active"       "race"         "fhx"          "imspLEVEL"    "SampleType"</code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\nNumber of levels in selected group\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code>## 
## Number of levels in selected group</code></pre>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps_ibd_phylo</span><span class="op">)</span><span class="op">$</span><span class="va">DiseaseState</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>## [1] 1 2 3 4</code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://david-barnett.github.io/microViz">microViz</a></span><span class="op">)</span></span>
<span><span class="va">ps_ibd_phylo</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/tax_fix.html">tax_fix</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/comp_barplot.html">comp_barplot</a></span><span class="op">(</span></span>
<span>    tax_level <span class="op">=</span> <span class="st">"Genus"</span>, n_taxa <span class="op">=</span> <span class="fl">15</span>, other_name <span class="op">=</span> <span class="st">"Other"</span>,</span>
<span>    taxon_renamer <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/stringr/man/str_remove.html">str_remove</a></span><span class="op">(</span><span class="va">x</span>, <span class="st">" [ae]t rel."</span><span class="op">)</span>,</span>
<span>    palette <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/distinct_palette.html">distinct_palette</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">15</span>, add <span class="op">=</span> <span class="st">"grey90"</span><span class="op">)</span>,</span>
<span>    merge_other <span class="op">=</span> <span class="cn">FALSE</span>, bar_outline_colour <span class="op">=</span> <span class="st">"darkgrey"</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/coord_flip.html">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/labs.html">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/theme.html">theme</a></span><span class="op">(</span>axis.text.y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>, axis.ticks.y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="02_explore_processed_data_files/figure-html/unnamed-chunk-4-1.png" width="960"></div>
</div>
<div id="dataset-ps_dietswap" class="section level2" number="8.4">
<h2>
<span class="header-section-number">8.4</span> Dataset: <code>ps_dietswap</code><a class="anchor" aria-label="anchor" href="#dataset-ps_dietswap"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ps_dietswap</span></span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 130 taxa and 222 samples ]
## sample_data() Sample Data:       [ 222 samples by 8 sample variables ]
## tax_table()   Taxonomy Table:    [ 130 taxa by 3 taxonomic ranks ]</code></pre>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\nSample Variables\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code>## 
## Sample Variables</code></pre>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_variables.html">sample_variables</a></span><span class="op">(</span><span class="va">ps_dietswap</span><span class="op">)</span></span></code></pre></div>
<pre><code>## [1] "subject"                "sex"                    "nationality"           
## [4] "group"                  "sample"                 "timepoint"             
## [7] "timepoint.within.group" "bmi_group"</code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\nNumber of levels in nationality\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code>## 
## Number of levels in nationality</code></pre>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps_dietswap</span><span class="op">)</span><span class="op">$</span><span class="va">nationality</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>## [1] 1 2</code></pre>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\nNumber of levels in group\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code>## 
## Number of levels in group</code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps_dietswap</span><span class="op">)</span><span class="op">$</span><span class="va">group</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>## [1] 1 2 3</code></pre>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/microbiomeMarker/man/run_lefse.html">run_lefse</a></span><span class="op">(</span></span>
<span>    <span class="va">ps_dietswap</span>,</span>
<span>    wilcoxon_cutoff <span class="op">=</span> <span class="fl">0.001</span>,</span>
<span>    group <span class="op">=</span> <span class="st">"nationality"</span>,</span>
<span>    taxa_rank <span class="op">=</span> <span class="st">"Phylum"</span>,</span>
<span>    transform <span class="op">=</span> <span class="st">"log10"</span>,</span>
<span>    kw_cutoff <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>    multigrp_strat <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    lda_cutoff <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/microbiomeMarker/man/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span>group <span class="op">=</span> <span class="st">"nationality"</span>, color <span class="op">=</span> <span class="st">"rainbow, 1:length(unique(df_dietswap$nationality))"</span><span class="op">)</span></span></code></pre></div>
<pre><code>## Warning in transform_log10(otu): OTU table contains zeroes. Using log10(1 + x)
## instead.

## Warning in transform_log10(otu): OTU table contains zeroes. Using log10(1 + x)
## instead.</code></pre>
<div class="inline-figure"><img src="02_explore_processed_data_files/figure-html/unnamed-chunk-6-1.png" width="672"></div>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://david-barnett.github.io/microViz">microViz</a></span><span class="op">)</span></span>
<span><span class="va">ps_dietswap</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/comp_barplot.html">comp_barplot</a></span><span class="op">(</span></span>
<span>    tax_level <span class="op">=</span> <span class="st">"Genus"</span>, n_taxa <span class="op">=</span> <span class="fl">15</span>, other_name <span class="op">=</span> <span class="st">"Other"</span>,</span>
<span>    taxon_renamer <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/stringr/man/str_remove.html">str_remove</a></span><span class="op">(</span><span class="va">x</span>, <span class="st">" [ae]t rel."</span><span class="op">)</span>,</span>
<span>    palette <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/distinct_palette.html">distinct_palette</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">15</span>, add <span class="op">=</span> <span class="st">"grey90"</span><span class="op">)</span>,</span>
<span>    merge_other <span class="op">=</span> <span class="cn">FALSE</span>, bar_outline_colour <span class="op">=</span> <span class="st">"darkgrey"</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/coord_flip.html">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="st">"nationality"</span>, nrow <span class="op">=</span> <span class="fl">1</span>, scales <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/labs.html">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/theme.html">theme</a></span><span class="op">(</span>axis.text.y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>, axis.ticks.y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="02_explore_processed_data_files/figure-html/dietswap-1.png" width="960"></div>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># ## Recoding values in phyloseq object</span></span>
<span><span class="co"># data("dietswap", package = "microbiome")</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># # create a couple of numerical variables to use as constraints or conditions</span></span>
<span><span class="co"># dietswap &lt;- dietswap %&gt;%</span></span>
<span><span class="co">#   ps_mutate(</span></span>
<span><span class="co">#     weight = recode(bmi_group, obese = 3, overweight = 2, lean = 1),</span></span>
<span><span class="co">#     female = if_else(sex == "female", true = 1, false = 0),</span></span>
<span><span class="co">#     african = if_else(nationality == "AFR", true = 1, false = 0)</span></span>
<span><span class="co">#   )</span></span>
<span><span class="co"># # add a couple of missing values to show how microViz handles missing data</span></span>
<span><span class="co"># sample_data(dietswap)$african[c(3, 4)] &lt;- NA</span></span></code></pre></div>
</div>
<div id="dataset-ps_caporaso" class="section level2" number="8.5">
<h2>
<span class="header-section-number">8.5</span> Dataset: <code>ps_caporaso</code><a class="anchor" aria-label="anchor" href="#dataset-ps_caporaso"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ps_caporaso</span></span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 3426 taxa and 34 samples ]
## sample_data() Sample Data:       [ 34 samples by 8 sample variables ]
## tax_table()   Taxonomy Table:    [ 3426 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 3426 tips and 3424 internal nodes ]</code></pre>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\nSample Variables\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code>## 
## Sample Variables</code></pre>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_variables.html">sample_variables</a></span><span class="op">(</span><span class="va">ps_caporaso</span><span class="op">)</span></span></code></pre></div>
<pre><code>## [1] "SampleType"               "Year"                    
## [3] "Month"                    "Day"                     
## [5] "Subject"                  "ReportedAntibioticUsage" 
## [7] "DaysSinceExperimentStart" "Description"</code></pre>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\nNumber of levels in selected group\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code>## 
## Number of levels in selected group</code></pre>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">df_caporaso</span><span class="op">$</span><span class="va">SampleType</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>## [1] 1 2 3 4</code></pre>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ps_caporaso</span> <span class="op">&lt;-</span> <span class="va">ps_caporaso</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span>  <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/tax_fix.html">tax_fix</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/microbiomeMarker/man/run_lefse.html">run_lefse</a></span><span class="op">(</span></span>
<span>    <span class="va">ps_caporaso</span>, </span>
<span>    wilcoxon_cutoff <span class="op">=</span> <span class="fl">0.001</span>,</span>
<span>    group <span class="op">=</span> <span class="st">"SampleType"</span>,</span>
<span>    taxa_rank <span class="op">=</span> <span class="st">"Class"</span>,</span>
<span>    transform <span class="op">=</span> <span class="st">"log10p"</span>,</span>
<span>    kw_cutoff <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>    multigrp_strat <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    lda_cutoff <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/microbiomeMarker/man/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span>group <span class="op">=</span> <span class="st">"SampleType"</span>, color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"green"</span>, <span class="st">"red"</span>, <span class="st">"grey"</span>, <span class="st">"magenta"</span>, <span class="st">"orange"</span>, <span class="st">"pink"</span>, <span class="st">"yellow"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="02_explore_processed_data_files/figure-html/ps_caporaso_heatmap-1.png" width="960"></div>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://david-barnett.github.io/microViz">microViz</a></span><span class="op">)</span></span>
<span><span class="va">ps_caporaso</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/tax_fix.html">tax_fix</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/comp_barplot.html">comp_barplot</a></span><span class="op">(</span></span>
<span>    tax_level <span class="op">=</span> <span class="st">"Genus"</span>, n_taxa <span class="op">=</span> <span class="fl">15</span>, other_name <span class="op">=</span> <span class="st">"Other"</span>,</span>
<span>    taxon_renamer <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/stringr/man/str_remove.html">str_remove</a></span><span class="op">(</span><span class="va">x</span>, <span class="st">" [ae]t rel."</span><span class="op">)</span>,</span>
<span>    palette <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/distinct_palette.html">distinct_palette</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">15</span>, add <span class="op">=</span> <span class="st">"grey90"</span><span class="op">)</span>,</span>
<span>    merge_other <span class="op">=</span> <span class="cn">FALSE</span>, bar_outline_colour <span class="op">=</span> <span class="st">"darkgrey"</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/coord_flip.html">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="st">"SampleType"</span>, nrow <span class="op">=</span> <span class="fl">1</span>, scales <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/labs.html">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="cn">NULL</span>, y <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/theme.html">theme</a></span><span class="op">(</span>axis.text.y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>, axis.ticks.y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="02_explore_processed_data_files/figure-html/unnamed-chunk-9-1.png" width="960"></div>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"caporaso"</span><span class="op">)</span></span>
<span><span class="va">caporaso_small</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_taxa-methods.html">subset_taxa</a></span><span class="op">(</span></span>
<span>    <span class="va">caporaso</span>,</span>
<span>    <span class="va">Phylum</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"p__Firmicutes"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">mm_lefse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/microbiomeMarker/man/run_lefse.html">run_lefse</a></span><span class="op">(</span></span>
<span>    <span class="va">caporaso_small</span>,</span>
<span>    wilcoxon_cutoff <span class="op">=</span> <span class="fl">0.0001</span>,</span>
<span>    group <span class="op">=</span> <span class="st">"SampleType"</span>,</span>
<span>    kw_cutoff <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>    multigrp_strat <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    lda_cutoff <span class="op">=</span> <span class="fl">4</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/microbiomeMarker/man/plot_cladogram.html">plot_cladogram</a></span><span class="op">(</span><span class="va">mm_lefse</span>, color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"green"</span>, <span class="st">"red"</span>, <span class="st">"magenta"</span>, <span class="st">"yellow"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="02_explore_processed_data_files/figure-html/ps_caporaso_cladogram-1.png" width="960"></div>
</div>
<div id="dataset-ps_kostic_crc" class="section level2" number="8.6">
<h2>
<span class="header-section-number">8.6</span> Dataset: <code>ps_kostic_crc</code><a class="anchor" aria-label="anchor" href="#dataset-ps_kostic_crc"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ps_kostic_crc</span></span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 2505 taxa and 177 samples ]
## sample_data() Sample Data:       [ 177 samples by 71 sample variables ]
## tax_table()   Taxonomy Table:    [ 2505 taxa by 7 taxonomic ranks ]</code></pre>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\nSample Variables\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code>## 
## Sample Variables</code></pre>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_variables.html">sample_variables</a></span><span class="op">(</span><span class="va">ps_kostic_crc</span><span class="op">)</span></span></code></pre></div>
<pre><code>##  [1] "X.SampleID"                    "BarcodeSequence"              
##  [3] "LinkerPrimerSequence"          "NECROSIS_PERCENT"             
##  [5] "TARGET_SUBFRAGMENT"            "ASSIGNED_FROM_GEO"            
##  [7] "EXPERIMENT_CENTER"             "TITLE"                        
##  [9] "RUN_PREFIX"                    "AGE"                          
## [11] "NORMAL_EQUIVALENT_PERCENT"     "FIBROBLAST_AND_VESSEL_PERCENT"
## [13] "DEPTH"                         "TREATMENT"                    
## [15] "AGE_AT_DIAGNOSIS"              "COMMON_NAME"                  
## [17] "HOST_COMMON_NAME"              "BODY_SITE"                    
## [19] "ELEVATION"                     "REPORTS_RECEIVED"             
## [21] "CEA"                           "PCR_PRIMERS"                  
## [23] "COLLECTION_DATE"               "ALTITUDE"                     
## [25] "ENV_BIOME"                     "SEX"                          
## [27] "PLATFORM"                      "RACE"                         
## [29] "BSP_DIAGNOSIS"                 "STUDY_CENTER"                 
## [31] "COUNTRY"                       "CHEMOTHERAPY"                 
## [33] "YEAR_OF_DEATH"                 "ETHNICITY"                    
## [35] "ANONYMIZED_NAME"               "TAXON_ID"                     
## [37] "SAMPLE_CENTER"                 "SAMP_SIZE"                    
## [39] "YEAR_OF_BIRTH"                 "ORIGINAL_DIAGNOSIS"           
## [41] "AGE_UNIT"                      "STUDY_ID"                     
## [43] "EXPERIMENT_DESIGN_DESCRIPTION" "Description_duplicate"        
## [45] "DIAGNOSIS"                     "BODY_HABITAT"                 
## [47] "SEQUENCING_METH"               "RUN_DATE"                     
## [49] "HISTOLOGIC_GRADE"              "LONGITUDE"                    
## [51] "ENV_MATTER"                    "TARGET_GENE"                  
## [53] "ENV_FEATURE"                   "KEY_SEQ"                      
## [55] "BODY_PRODUCT"                  "TUMOR_PERCENT"                
## [57] "LIBRARY_CONSTRUCTION_PROTOCOL" "REGION"                       
## [59] "RUN_CENTER"                    "TUMOR_TYPE"                   
## [61] "BSP_NOTES"                     "RADIATION_THERAPY"            
## [63] "INFLAMMATION_PERCENT"          "HOST_SUBJECT_ID"              
## [65] "PC3"                           "LATITUDE"                     
## [67] "OSH_DIAGNOSIS"                 "STAGE"                        
## [69] "PRIMARY_DISEASE"               "HOST_TAXID"                   
## [71] "Description"</code></pre>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"\nNumber of levels in DIAGNOSIS\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code>## 
## Number of levels in DIAGNOSIS</code></pre>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps_kostic_crc</span><span class="op">)</span><span class="op">$</span><span class="va">DIAGNOSIS</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>## [1] 1 2</code></pre>
<div class="sourceCode" id="cb87"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/microbiomeMarker/man/run_lefse.html">run_lefse</a></span><span class="op">(</span></span>
<span>    <span class="va">ps_kostic_crc</span>,</span>
<span>    wilcoxon_cutoff <span class="op">=</span> <span class="fl">0.001</span>,</span>
<span>    group <span class="op">=</span> <span class="st">"DIAGNOSIS"</span>,</span>
<span>    taxa_rank <span class="op">=</span> <span class="st">"Class"</span>,</span>
<span>    transform <span class="op">=</span> <span class="st">"log10p"</span>,</span>
<span>    kw_cutoff <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>    multigrp_strat <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    lda_cutoff <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/microbiomeMarker/man/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span>group <span class="op">=</span> <span class="st">"DIAGNOSIS"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="02_explore_processed_data_files/figure-html/ps_kostic_crc_heatmap-1.png" width="960"></div>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">kostic_crc_small</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/subset_taxa-methods.html">subset_taxa</a></span><span class="op">(</span></span>
<span>    <span class="va">ps_kostic_crc</span>,</span>
<span>    <span class="va">Phylum</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Firmicutes"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">mm_lefse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/microbiomeMarker/man/run_lefse.html">run_lefse</a></span><span class="op">(</span></span>
<span>    <span class="va">kostic_crc_small</span>,</span>
<span>    wilcoxon_cutoff <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>    group <span class="op">=</span> <span class="st">"DIAGNOSIS"</span>,</span>
<span>    kw_cutoff <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>    multigrp_strat <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    lda_cutoff <span class="op">=</span> <span class="fl">4</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/microbiomeMarker/man/plot_cladogram.html">plot_cladogram</a></span><span class="op">(</span><span class="va">mm_lefse</span>, color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"darkgreen"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="02_explore_processed_data_files/figure-html/ps_kostic_crc_cladogram-1.png" width="960"></div>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">htmp</span> <span class="op">&lt;-</span> <span class="va">ps_dietswap</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/ps_mutate.html">ps_mutate</a></span><span class="op">(</span>nationality <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">nationality</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/tax_transform.html">tax_transform</a></span><span class="op">(</span><span class="st">"log2"</span>, add <span class="op">=</span> <span class="fl">1</span>, chain <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/comp_heatmap.html">comp_heatmap</a></span><span class="op">(</span></span>
<span>    taxa <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/tax_top.html">tax_top</a></span><span class="op">(</span><span class="va">ps_dietswap</span>, n <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>, grid_col <span class="op">=</span> <span class="cn">NA</span>, name <span class="op">=</span> <span class="st">"Log2p"</span>,</span>
<span>    taxon_renamer <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/stringr/man/str_remove.html">str_remove</a></span><span class="op">(</span><span class="va">x</span>, <span class="st">" [ae]t rel."</span><span class="op">)</span>,</span>
<span>    colors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/heat_palette.html">heat_palette</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="fu">viridis</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/viridisLite/man/viridis.html">turbo</a></span><span class="op">(</span><span class="fl">11</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    row_names_side <span class="op">=</span> <span class="st">"left"</span>, row_dend_side <span class="op">=</span> <span class="st">"right"</span>, sample_side <span class="op">=</span> <span class="st">"bottom"</span>,</span>
<span>    sample_anno <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/sampleAnnotation.html">sampleAnnotation</a></span><span class="op">(</span></span>
<span>      Nationality <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/anno_sample_cat.html">anno_sample_cat</a></span><span class="op">(</span></span>
<span>        var <span class="op">=</span> <span class="st">"nationality"</span>, col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>AAM <span class="op">=</span> <span class="st">"grey35"</span>, AFR <span class="op">=</span> <span class="st">"grey85"</span><span class="op">)</span>,</span>
<span>        box_col <span class="op">=</span> <span class="cn">NA</span>, legend_title <span class="op">=</span> <span class="st">"Nationality"</span>, size <span class="op">=</span> <span class="fu">grid</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">4</span>, <span class="st">"mm"</span><span class="op">)</span></span>
<span>      <span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="fu">ComplexHeatmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/draw-dispatch.html">draw</a></span><span class="op">(</span></span>
<span>  object <span class="op">=</span> <span class="va">htmp</span>, annotation_legend_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/attr.html">attr</a></span><span class="op">(</span><span class="va">htmp</span>, <span class="st">"AnnoLegends"</span><span class="op">)</span>,</span>
<span>  merge_legends <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="02_explore_processed_data_files/figure-html/unnamed-chunk-10-1.png" width="960"></div>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># perform ordination</span></span>
<span><span class="va">unconstrained_aitchison_pca</span> <span class="op">&lt;-</span> <span class="va">ps_dietswap</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/tax_filter.html">tax_filter</a></span><span class="op">(</span>min_prevalence <span class="op">=</span> <span class="fl">0.1</span>, tax_level <span class="op">=</span> <span class="st">"Genus"</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/tax_agg.html">tax_agg</a></span><span class="op">(</span><span class="st">"Family"</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/tax_transform.html">tax_transform</a></span><span class="op">(</span><span class="st">"clr"</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/ord_calc.html">ord_calc</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code>## Proportional min_prevalence given: 0.1 --&gt; min 23/222 samples.</code></pre>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; Proportional min_prevalence given: 0.1 --&gt; min 23/222 samples.</span></span>
<span><span class="co"># ord_calc will automatically infer you want a "PCA" here</span></span>
<span><span class="co"># specify explicitly with method = "PCA", or you can pick another method</span></span>
<span></span>
<span><span class="co"># create plot</span></span>
<span><span class="va">pca_plot</span> <span class="op">&lt;-</span> <span class="va">unconstrained_aitchison_pca</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/ord_plot.html">ord_plot</a></span><span class="op">(</span></span>
<span>    plot_taxa <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">6</span>, colour <span class="op">=</span> <span class="st">"bmi_group"</span>, size <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>    tax_vec_length <span class="op">=</span> <span class="fl">0.325</span>,</span>
<span>    tax_lab_style <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/Ordination-labels.html">tax_lab_style</a></span><span class="op">(</span>max_angle <span class="op">=</span> <span class="fl">90</span>, aspect_ratio <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>    auto_caption <span class="op">=</span> <span class="fl">8</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># customise plot</span></span>
<span><span class="va">customised_plot</span> <span class="op">&lt;-</span> <span class="va">pca_plot</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/stat_ellipse.html">stat_ellipse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/aes.html">aes</a></span><span class="op">(</span>linetype <span class="op">=</span> <span class="va">bmi_group</span>, colour <span class="op">=</span> <span class="va">bmi_group</span><span class="op">)</span>, linewidth <span class="op">=</span> <span class="fl">0.3</span><span class="op">)</span> <span class="op">+</span> <span class="co"># linewidth not size, since ggplot 3.4.0</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/scale_brewer.html">scale_colour_brewer</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set1"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/coord_fixed.html">coord_fixed</a></span><span class="op">(</span>ratio <span class="op">=</span> <span class="fl">0.5</span>, clip <span class="op">=</span> <span class="st">"off"</span><span class="op">)</span> <span class="co"># makes rotated labels align correctly</span></span>
<span></span>
<span><span class="co"># show plot</span></span>
<span><span class="va">customised_plot</span></span></code></pre></div>
<div class="inline-figure"><img src="02_explore_processed_data_files/figure-html/unconstrained_aitchison_pca-1.png" width="672"></div>
<div class="sourceCode" id="cb93"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Test for significance</span></span>
<span><span class="co"># calculate distances</span></span>
<span><span class="va">aitchison_dists</span> <span class="op">&lt;-</span> <span class="va">ps_dietswap</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/tax_filter.html">tax_filter</a></span><span class="op">(</span>min_prevalence <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/tax_transform.html">tax_transform</a></span><span class="op">(</span><span class="st">"identity"</span>, rank <span class="op">=</span> <span class="st">"Family"</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/dist_calc.html">dist_calc</a></span><span class="op">(</span><span class="st">"aitchison"</span><span class="op">)</span></span></code></pre></div>
<pre><code>## Proportional min_prevalence given: 0.1 --&gt; min 23/222 samples.</code></pre>
<div class="sourceCode" id="cb95"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; Proportional min_prevalence given: 0.1 --&gt; min 23/222 samples.</span></span>
<span></span>
<span><span class="co"># the more permutations you request, the longer it takes</span></span>
<span><span class="co"># but also the more stable and precise your p-values become</span></span>
<span><span class="va">aitchison_perm</span> <span class="op">&lt;-</span> <span class="va">aitchison_dists</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/dist_permanova.html">dist_permanova</a></span><span class="op">(</span></span>
<span>    seed <span class="op">=</span> <span class="fl">1234</span>, <span class="co"># for set.seed to ensure reproducibility of random process</span></span>
<span>    n_processes <span class="op">=</span> <span class="fl">1</span>, n_perms <span class="op">=</span> <span class="fl">99</span>, <span class="co"># you should use at least 999!</span></span>
<span>    variables <span class="op">=</span> <span class="st">"bmi_group"</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<pre><code>## 2024-02-21 15:01:26.263041 - Starting PERMANOVA with 99 perms with 1 processes</code></pre>
<pre><code>## 2024-02-21 15:01:26.325414 - Finished PERMANOVA</code></pre>
<div class="sourceCode" id="cb98"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; 2023-04-03 15:50:14 - Starting PERMANOVA with 99 perms with 1 processes</span></span>
<span><span class="co">#&gt; 2023-04-03 15:50:14 - Finished PERMANOVA</span></span>
<span></span>
<span><span class="co"># view the permanova results</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/microViz/man/psExtra-accessors.html">perm_get</a></span><span class="op">(</span><span class="va">aitchison_perm</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code>##            Df  SumOfSqs         R2        F Pr(&gt;F)
## bmi_group   2  104.0678 0.04177157 4.773379   0.01
## Residual  219 2387.2862 0.95822843       NA     NA
## Total     221 2491.3540 1.00000000       NA     NA</code></pre>
<div class="sourceCode" id="cb100"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt;            Df  SumOfSqs         R2        F Pr(&gt;F)</span></span>
<span><span class="co">#&gt; bmi_group   2  104.0678 0.04177157 4.773379   0.01</span></span>
<span><span class="co">#&gt; Residual  219 2387.2862 0.95822843       NA     NA</span></span>
<span><span class="co">#&gt; Total     221 2491.3540 1.00000000       NA     NA</span></span>
<span></span>
<span><span class="co"># view the info stored about the distance calculation</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/microViz/man/psExtra-accessors.html">info_get</a></span><span class="op">(</span><span class="va">aitchison_perm</span><span class="op">)</span></span></code></pre></div>
<pre><code>## psExtra info:
## tax_agg = "Family" tax_trans = "identity" dist_method = "aitchison"</code></pre>
<div class="sourceCode" id="cb102"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; psExtra info:</span></span>
<span><span class="co">#&gt; tax_agg = "Family" tax_trans = "identity" dist_method = "aitchison"</span></span></code></pre></div>
<div class="sourceCode" id="cb103"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># perm2 &lt;- aitchison_dists %&gt;%</span></span>
<span><span class="co">#   dist_permanova(variables = c("weight", "african", "sex"), seed = 321)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># perm2 %&gt;%</span></span>
<span><span class="co">#   ord_calc(constraints = c("weight", "african"), conditions = "female") %&gt;%</span></span>
<span><span class="co">#   ord_plot(</span></span>
<span><span class="co">#     colour = "nationality", size = 2.5, alpha = 0.35,</span></span>
<span><span class="co">#     auto_caption = 7,</span></span>
<span><span class="co">#     constraint_vec_length = 1,</span></span>
<span><span class="co">#     constraint_vec_style = vec_constraint(1.5, colour = "grey15"),</span></span>
<span><span class="co">#     constraint_lab_style = constraint_lab_style(</span></span>
<span><span class="co">#       max_angle = 90, size = 3, aspect_ratio = 0.35, colour = "black"</span></span>
<span><span class="co">#     )</span></span>
<span><span class="co">#   ) +</span></span>
<span><span class="co">#   stat_ellipse(aes(colour = nationality), linewidth = 0.2) + # linewidth not size since ggplot 3.4.0</span></span>
<span><span class="co">#   scale_color_brewer(palette = "Set1") +</span></span>
<span><span class="co">#   coord_fixed(ratio = 0.35, clip = "off") +</span></span>
<span><span class="co">#   theme(legend.position = c(0.9, 0.1), legend.background = element_rect())</span></span></code></pre></div>
<div class="sourceCode" id="cb104"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Correlation heatmap</span></span>
<span><span class="co"># set up the data with numerical variables and filter to top taxa</span></span>
<span><span class="va">psq</span> <span class="op">&lt;-</span> <span class="va">ps_dietswap</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/ps_mutate.html">ps_mutate</a></span><span class="op">(</span></span>
<span>    weight <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/recode.html">recode</a></span><span class="op">(</span><span class="va">bmi_group</span>, obese <span class="op">=</span> <span class="fl">3</span>, overweight <span class="op">=</span> <span class="fl">2</span>, lean <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>    female <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/if_else.html">if_else</a></span><span class="op">(</span><span class="va">sex</span> <span class="op">==</span> <span class="st">"female"</span>, true <span class="op">=</span> <span class="fl">1</span>, false <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>,</span>
<span>    african <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/if_else.html">if_else</a></span><span class="op">(</span><span class="va">nationality</span> <span class="op">==</span> <span class="st">"AFR"</span>, true <span class="op">=</span> <span class="fl">1</span>, false <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/tax_filter.html">tax_filter</a></span><span class="op">(</span></span>
<span>    tax_level <span class="op">=</span> <span class="st">"Genus"</span>, min_prevalence <span class="op">=</span> <span class="fl">1</span> <span class="op">/</span> <span class="fl">10</span>, min_sample_abundance <span class="op">=</span> <span class="fl">1</span> <span class="op">/</span> <span class="fl">10</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/tax_transform.html">tax_transform</a></span><span class="op">(</span><span class="st">"identity"</span>, rank <span class="op">=</span> <span class="st">"Genus"</span><span class="op">)</span></span></code></pre></div>
<pre><code>## Proportional min_prevalence given: 0.1 --&gt; min 23/222 samples.</code></pre>
<div class="sourceCode" id="cb106"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#&gt; Proportional min_prevalence given: 0.1 --&gt; min 23/222 samples.</span></span>
<span></span>
<span><span class="co"># randomly select 30 taxa from the 50 most abundant taxa (just for an example)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">taxa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/microViz/man/tax_top.html">tax_top</a></span><span class="op">(</span><span class="va">psq</span>, n <span class="op">=</span> <span class="fl">50</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></span>
<span><span class="co"># actually draw the heatmap</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/microViz/man/cor_heatmap.html">cor_heatmap</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">psq</span>, taxa <span class="op">=</span> <span class="va">taxa</span>,</span>
<span>  taxon_renamer <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/stringr/man/str_remove.html">str_remove</a></span><span class="op">(</span><span class="va">x</span>, <span class="st">" [ae]t rel."</span><span class="op">)</span>,</span>
<span>  tax_anno <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/taxAnnotation.html">taxAnnotation</a></span><span class="op">(</span></span>
<span>    Prev. <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/anno_tax_prev.html">anno_tax_prev</a></span><span class="op">(</span>undetected <span class="op">=</span> <span class="fl">50</span><span class="op">)</span>,</span>
<span>    Log2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/microViz/man/anno_tax_box.html">anno_tax_box</a></span><span class="op">(</span>undetected <span class="op">=</span> <span class="fl">50</span>, trans <span class="op">=</span> <span class="st">"log2"</span>, zero_replace <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="02_explore_processed_data_files/figure-html/unnamed-chunk-13-1.png" width="672"></div>
</div>
<div id="plot-ggtree-cladogram" class="section level2" number="8.7">
<h2>
<span class="header-section-number">8.7</span> Plot ggtree cladogram<a class="anchor" aria-label="anchor" href="#plot-ggtree-cladogram"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># library(ggtree)</span></span>
<span></span>
<span><span class="co"># ggtree(</span></span>
<span><span class="co">#   ps_tree,</span></span>
<span><span class="co">#   mapping = NULL,</span></span>
<span><span class="co">#   layout = "circular",</span></span>
<span><span class="co">#   open.angle = 0,</span></span>
<span><span class="co">#   mrsd = NULL,</span></span>
<span><span class="co">#   as.Date = FALSE,</span></span>
<span><span class="co">#   yscale = "none",</span></span>
<span><span class="co">#   yscale_mapping = NULL,</span></span>
<span><span class="co">#   ladderize = TRUE,</span></span>
<span><span class="co">#   right = FALSE,</span></span>
<span><span class="co">#   branch.length = "branch.length",</span></span>
<span><span class="co">#   root.position = 0,</span></span>
<span><span class="co">#   xlim = NULL</span></span>
<span><span class="co"># )</span></span></code></pre></div>
<div class="sourceCode" id="cb108"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># External data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"~/Dropbox/MICROBIOME/imap-data-processing/data/external/external_ps_objects.rda"</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code>## Loading objects:
##   df_GlobalPatterns
##   df_ibd_phylo
##   df_dietswap
##   df_caporaso
##   df_kostic_crc
##   ps_GlobalPatterns
##   ps_ibd_phylo
##   ps_dietswap
##   ps_caporaso
##   ps_kostic_crc</code></pre>
<div class="sourceCode" id="cb110"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vegandevs/vegan">vegan</a></span><span class="op">)</span></span></code></pre></div>
<pre><code>## Loading required package: permute</code></pre>
<pre><code>## Loading required package: lattice</code></pre>
<pre><code>## This is vegan 2.6-4</code></pre>
<pre><code>## 
## Attaching package: 'vegan'</code></pre>
<pre><code>## The following object is masked from 'package:microbiome':
## 
##     diversity</code></pre>
<div class="sourceCode" id="cb116"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># # Mothur</span></span>
<span><span class="co"># load("~/Dropbox/MICROBIOME/imap-data-processing/data/mothur/mothur_phyloseq_objects.rda", verbose = TRUE)</span></span>
<span><span class="co"># composite &lt;- read_csv("~/Dropbox/MICROBIOME/imap-data-processing/data/mothur/mothur_composite.csv", show_col_types = FALSE)</span></span>
<span>  <span class="co"># separate(Sample, into = c("animal", "day"), sep = "D", remove = FALSE, convert = TRUE) %&gt;%</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># taxa &lt;- sample(tax_top(ps_dietswap, n = 30), size = 20)</span></span></code></pre></div>
</div>
<div id="getting-otutable-from-phyloseq" class="section level2" number="8.8">
<h2>
<span class="header-section-number">8.8</span> Getting <code>otutable</code> from phyloseq<a class="anchor" aria-label="anchor" href="#getting-otutable-from-phyloseq"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb117"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vegandevs/vegan">vegan</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217">phyloseq</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">ps</span> <span class="op">&lt;-</span> <span class="va">ps_dietswap</span></span>
<span></span>
<span><span class="va">otutable</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/psmelt.html">psmelt</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/group_by.html">group_by</a></span><span class="op">(</span><span class="va">Sample</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/mutate.html">mutate</a></span><span class="op">(</span>N <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">Abundance</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/group_by.html">ungroup</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="co"># filter(N &gt;= min(otutable$N)) %&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/select.html">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">N</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/tidyr/man/pivot_wider.html">pivot_wider</a></span><span class="op">(</span>names_from<span class="op">=</span><span class="st">"OTU"</span>, values_from<span class="op">=</span><span class="st">"Abundance"</span>, values_fill<span class="op">=</span><span class="fl">0</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/tibble/man/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"Sample"</span><span class="op">)</span></span></code></pre></div>
</div>
<div id="getting-bray-curtis-distances" class="section level2" number="8.9">
<h2>
<span class="header-section-number">8.9</span> Getting Bray-<code>Curtis</code> distances<a class="anchor" aria-label="anchor" href="#getting-bray-curtis-distances"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb118"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bray</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/avgdist.html">avgdist</a></span><span class="op">(</span><span class="va">otutable</span>, dmethod<span class="op">=</span><span class="st">"bray"</span>, sample<span class="op">=</span><span class="fl">1776</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/tibble/man/as_tibble.html">as_tibble</a></span><span class="op">(</span>rownames <span class="op">=</span> <span class="st">"A"</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/tidyr/man/pivot_longer.html">pivot_longer</a></span><span class="op">(</span><span class="op">-</span><span class="va">A</span>, names_to<span class="op">=</span><span class="st">"B"</span>, values_to<span class="op">=</span><span class="st">"distances"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">bray</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">A</span>, y<span class="op">=</span><span class="va">B</span>, fill<span class="op">=</span><span class="va">distances</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/geom_tile.html">geom_tile</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="02_explore_processed_data_files/figure-html/braydist-1.png" width="672"></div>
</div>
<div id="getting-jaccard-distances" class="section level2" number="8.10">
<h2>
<span class="header-section-number">8.10</span> Getting <code>Jaccard</code> distances<a class="anchor" aria-label="anchor" href="#getting-jaccard-distances"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb119"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">jaccard</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/avgdist.html">avgdist</a></span><span class="op">(</span><span class="va">otutable</span>, dmethod<span class="op">=</span><span class="st">"jaccard"</span>, sample<span class="op">=</span><span class="fl">1776</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/tibble/man/as_tibble.html">as_tibble</a></span><span class="op">(</span>rownames <span class="op">=</span> <span class="st">"A"</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/tidyr/man/pivot_longer.html">pivot_longer</a></span><span class="op">(</span><span class="op">-</span><span class="va">A</span>, names_to<span class="op">=</span><span class="st">"B"</span>, values_to<span class="op">=</span><span class="st">"distances"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">jaccard</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">A</span>, y<span class="op">=</span><span class="va">B</span>, fill<span class="op">=</span><span class="va">distances</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/geom_tile.html">geom_tile</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="02_explore_processed_data_files/figure-html/jaccarddist-1.png" width="672"></div>
<div class="sourceCode" id="cb120"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Some params</span></span>
<span></span>
<span><span class="va">labels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/tibble/man/tibble.html">tibble</a></span><span class="op">(</span></span>
<span>  x<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">50</span>, <span class="fl">190</span><span class="op">)</span>,</span>
<span>  y<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">190</span>, <span class="fl">30</span><span class="op">)</span>,</span>
<span>  label<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Bray-Curtis"</span>, <span class="st">"Jaccard"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/mutate-joins.html">inner_join</a></span><span class="op">(</span><span class="va">bray</span>, <span class="va">jaccard</span>, by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"B"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/select.html">select</a></span><span class="op">(</span><span class="va">A</span>, <span class="va">B</span>, bray<span class="op">=</span><span class="va">distances.x</span>, jaccard<span class="op">=</span><span class="va">distances.y</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/mutate.html">mutate</a></span><span class="op">(</span>distances <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/if_else.html">if_else</a></span><span class="op">(</span><span class="va">A</span> <span class="op">&lt;</span> <span class="va">B</span>, <span class="va">bray</span>, <span class="va">jaccard</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html">%&gt;%</a></span></span>
<span>  <span class="co"># mutate(A = fct_reorder(as.character(A), A),</span></span>
<span>  <span class="co">#        B = fct_reorder(as.character(B), B)) %&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">A</span>, y<span class="op">=</span><span class="va">B</span>, fill<span class="op">=</span><span class="va">distances</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/geom_tile.html">geom_tile</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/geom_text.html">geom_text</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">labels</span>, <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, y<span class="op">=</span><span class="va">y</span>, label<span class="op">=</span><span class="va">label</span><span class="op">)</span>, inherit.aes<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>            size<span class="op">=</span><span class="fl">10</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/scale_gradient.html">scale_fill_gradient</a></span><span class="op">(</span>low<span class="op">=</span><span class="st">"#FF0000"</span>, high<span class="op">=</span><span class="st">"#FFFFFF"</span>, name<span class="op">=</span><span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/labs.html">labs</a></span><span class="op">(</span>x<span class="op">=</span><span class="st">""</span>, y<span class="op">=</span><span class="st">""</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/ggtheme.html">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/theme.html">theme</a></span><span class="op">(</span>axis.line<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.ticks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.text <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/element.html">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">8</span><span class="op">)</span>,</span>
<span>        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/element.html">element_text</a></span><span class="op">(</span>hjust<span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/element.html">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="02_explore_processed_data_files/figure-html/jaccarddist-2.png" width="672"></div>

</div>
</div>



  <div class="chapter-nav">
<div class="prev"><a href="conclusion.html"><span class="header-section-number">7</span> Conclusion</a></div>
<div class="next"><a href="imap-github-repos.html"><span class="header-section-number">A</span> IMAP GitHub Repos</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#exploration-of-processded-data"><span class="header-section-number">8</span> Exploration of processded data</a></li>
<li><a class="nav-link" href="#import-data"><span class="header-section-number">8.1</span> Import data</a></li>
<li><a class="nav-link" href="#dataset-ps_globalpatterns"><span class="header-section-number">8.2</span> Dataset: ps_GlobalPatterns</a></li>
<li><a class="nav-link" href="#dataset-ps_ibd_phylo"><span class="header-section-number">8.3</span> Dataset: ps_ibd_phylo</a></li>
<li><a class="nav-link" href="#dataset-ps_dietswap"><span class="header-section-number">8.4</span> Dataset: ps_dietswap</a></li>
<li><a class="nav-link" href="#dataset-ps_caporaso"><span class="header-section-number">8.5</span> Dataset: ps_caporaso</a></li>
<li><a class="nav-link" href="#dataset-ps_kostic_crc"><span class="header-section-number">8.6</span> Dataset: ps_kostic_crc</a></li>
<li><a class="nav-link" href="#plot-ggtree-cladogram"><span class="header-section-number">8.7</span> Plot ggtree cladogram</a></li>
<li><a class="nav-link" href="#getting-otutable-from-phyloseq"><span class="header-section-number">8.8</span> Getting otutable from phyloseq</a></li>
<li><a class="nav-link" href="#getting-bray-curtis-distances"><span class="header-section-number">8.9</span> Getting Bray-Curtis distances</a></li>
<li><a class="nav-link" href="#getting-jaccard-distances"><span class="header-section-number">8.10</span> Getting Jaccard distances</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/tmbuza/imap-data-exploration.git/blob/gh-pages/02_explore_processed_data.Rmd">View source <i class="fas fa-air-freshener"></i></a></li>
          <li><a id="book-edit" href="https://github.com/tmbuza/imap-data-exploration.git/edit/gh-pages/02_explore_processed_data.Rmd">Edit this page <i class="fas fa-air-freshener"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container">
    <div class="row">
      <div class="col-12 col-md-6 mt-3">
        <p><strong><u><a class="text-light" href="https://microbiome.complexdatainsights.com"><big>IMAP:</big> Integrated Microbiome Analysis Pipelines</a></u></strong>: End-to-End Practical User Guides Using Integrated Approaches</p>
<br><p>Authored by Teresia Mrema-Buza, Last updated: <b>2024-02-21</b>.</p>
<br>
</div>
      <div class="col-12 col-md-6 mt-3">
        <p>Note: This practical guide was created using the <a class="text-light" href="https://bookdown.org/yihui/bookdown/">bookdown</a> R package, styled with the modern bs4_book format.</p>
      </div>

    </div>
  </div>
</footer>
</body>
</html>
